worlds/celeste_open_world/__init__.py

```diff
@@ -0,0 +1,351 @@
+    item_name_groups: dict[str, list[str]] = generate_item_groups()
+
+
+    # Instance Data
+    madeline_one_dash_hair_color: int
+    madeline_two_dash_hair_color: int
+    madeline_no_dash_hair_color: int
+    madeline_feather_hair_color: int
+
+    active_levels: set[str]
+    active_items: set[str]
+
+
+    def generate_early(self) -> None:
+        if not self.player_name.isascii():
+            raise RuntimeError(f"Invalid player_name {self.player_name} for game {self.game}. Name must be ascii.")
+
+        resolve_options(self)
+
+        self.goal_area: str = goal_area_option_to_name[self.options.goal_area.value]
+
+        self.active_levels = {"0a", "1a", "2a", "3a", "4a", "5a", "6a", "7a", "8a"}
+        if self.options.include_core:
+            self.active_levels.add("9a")
+        if self.options.include_farewell >= 1:
+            self.active_levels.add("10a")
+        if self.options.include_farewell == 2:
+            self.active_levels.add("10b")
+        if self.options.include_b_sides:
+            self.active_levels.update({"1b", "2b", "3b", "4b", "5b", "6b", "7b"})
+            if self.options.include_core:
+                self.active_levels.add("9b")
+        if self.options.include_c_sides:
+            self.active_levels.update({"1c", "2c", "3c", "4c", "5c", "6c", "7c"})
+            if self.options.include_core:
+                self.active_levels.add("9c")
+
+        self.active_levels.add(self.goal_area)
+        if self.goal_area == "10c":
+            self.active_levels.add("10b")
+            self.active_levels.add("10a")
+        elif self.goal_area == "10b":
+            self.active_levels.add("10a")
+
+        self.active_items = set()
+        for level in self.active_levels:
+            self.active_items.update(level_item_lists[level])
+
+
+    def create_regions(self) -> None:
+        from .Locations import create_regions_and_locations
+
+        create_regions_and_locations(self)
+
+
+    def create_item(self, name: str, force_useful: bool = False) -> CelesteItem:
+        item_data_table = generate_item_data_table()
+
+        if name == ItemName.strawberry and force_useful:
+            return CelesteItem(name, ItemClassification.useful, item_data_table[name].code, self.player)
+        elif name in item_data_table:
+            return CelesteItem(name, item_data_table[name].type, item_data_table[name].code, self.player)
+        else:
+            return CelesteItem(name, ItemClassification.progression, None, self.player)
+
+    def create_items(self) -> None:
+        item_pool: list[CelesteItem] = []
+
+        location_count: int = len(self.get_locations())
+        goal_area_location_count: int = sum(goal_area_option_to_display_name[self.options.goal_area] in loc.name for loc in self.get_locations())
+
+        # Goal Items
+        goal_item_loc: Location = self.multiworld.get_location(goal_area_to_location_name[self.goal_area], self.player)
```

Comment:
```suggestion
        goal_item_loc: Location = self.get_location(goal_area_to_location_name[self.goal_area])
```
just a cleaner new-ish helper (and this goes for the other similar lines in the section obv)